crates/pallet-transporter/src/lib.rs

```diff
@@ -0,0 +1,296 @@
+    /// Unique identity of domain.
+    pub domain_id: DomainId,
+    /// Unique account on domain.
+    pub account_id: AccountId,
+}
+
+/// Transfer of funds from one domain to another.
+#[derive(Debug, Encode, Decode, Clone, Eq, PartialEq, TypeInfo)]
+pub struct Transfer<DomainId, AccountId, Balance> {
+    /// Amount being transferred between entities.
+    pub amount: Balance,
+    /// Sender location of the transfer.
+    pub sender: Location<DomainId, AccountId>,
+    /// Receiver location of the transfer.
+    pub receiver: Location<DomainId, AccountId>,
+}
+
+/// Balance type used by the pallet.
+pub(crate) type BalanceOf<T> =
+    <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+type MessageIdOf<T> =
+    <<T as Config>::Sender as sp_messenger::endpoint::Sender<<T as Config>::DomainId>>::MessageId;
+
+#[frame_support::pallet]
+mod pallet {
+    use crate::{BalanceOf, Location, MessageIdOf, Transfer};
+    use codec::{Decode, Encode};
+    use frame_support::pallet_prelude::*;
+    use frame_support::traits::{Currency, ExistenceRequirement, WithdrawReasons};
+    use frame_system::pallet_prelude::*;
+    use sp_messenger::endpoint::{
+        Endpoint, EndpointHandler as EndpointHandlerT, EndpointId, EndpointRequest,
+        EndpointResponse, Sender,
+    };
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// Event type for this pallet.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// Domain ID uniquely identifies a Domain.
+        type DomainId: Parameter + Member + Default + Copy + MaxEncodedLen;
+
+        /// Gets the domain_id of the current execution environment.
+        type SelfDomainId: Get<Self::DomainId>;
+
+        /// Gets the endpoint_id of the this pallet in a given execution environment.
+        type SelfEndpointId: Get<EndpointId>;
+
+        /// Currency used by this pallet.
+        type Currency: Currency<Self::AccountId>;
+
+        /// Sender used to transfer funds.
+        type Sender: Sender<Self::DomainId>;
+    }
+
+    /// Pallet transporter to move funds between domains.
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    /// All the outgoing transfers on this execution environment.
+    #[pallet::storage]
+    #[pallet::getter(fn outgoing_transfers)]
+    pub(super) type OutgoingTransfers<T: Config> = StorageDoubleMap<
+        _,
+        Identity,
+        T::DomainId,
+        Identity,
+        MessageIdOf<T>,
+        Transfer<T::DomainId, T::AccountId, BalanceOf<T>>,
+        OptionQuery,
+    >;
+
+    /// Events emitted by pallet-transporter.
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Emits when there is a new outgoing transfer.
+        OutgoingTransferInitiated {
```

Comment:
Personally, I'd probably use `Out` and `In` like `TransferOutInitiated` which are seemingly more concise(fewer characters). Not critical, just a thought.

Reply:
Okay. Will leave it as it for now :-D